File: app/Models/Registration.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Registration extends Model
{
    protected $fillable = [
        'event_id',
        'first_name',
        'last_name',
        'email',
        'registered_at',
    ];

    protected $hidden = ['updated_at', 'created_at'];

    protected function serializeDate(\DateTimeInterface $date): string
    {
        return $date->format('Y-m-d\TH:i:s\Z');
    }

    public function event()
    {
        return $this->belongsTo(Event::class);
    }

    // Formater en camelCase pour la réponse JSON
    public function toArray(): array
    {
        return [
            'id'           => $this->id,
            'eventId'      => $this->event_id,
            'firstName'    => $this->first_name,
            'lastName'     => $this->last_name,
            'email'        => $this->email,
            'registeredAt' => $this->registered_at,
        ];
    }
}

File: mobile/lib/models/registration.dart
class Registration {
  final int id;
  final int eventId;
  final String firstName;
  final String lastName;
  final String email;
  final String registeredAt;

  Registration({
    required this.id,
    required this.eventId,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.registeredAt,
  });

  String get fullName => '$firstName $lastName';

  factory Registration.fromJson(Map<String, dynamic> json) => Registration(
    id:           json['id'],
    eventId:      json['eventId'],
    firstName:    json['firstName'],
    lastName:     json['lastName'],
    email:        json['email'],
    registeredAt: json['registeredAt'],
  );
}
